### ISSUE TITLE
Animate view navigation in single-activity apps

### ISSUE THREAD

**Issue by FunnyDevs** [zserge/anvil#38]
Hi! i would like to try your library but before coding i would like to know if is possibile to animate navigation between RenderableViews. I haven't found any reference in readme. Thanks

**Comment by zserge** [comment]
Hi, the short answer is: Anvil can't animate your screen transitions, but Android SDK can.

Anvil itself is just a tiny wrapper on top of SDK view classes and methods to make them reactive. So everything you could do with android Views should be possible to do with Anvil as well. As such, RenderableView is just a regular FrameLayout in all its simplicity.

So, how would you animate your FrameLayout screens in your single-activity app if you were not using Anvil?

The first option could be to set layout transitions to the "root" view group (http://developer.android.com/training/animation/layout.html) so that it would animate inner layouts (screens) whenever your add or remove them. If your "view router" is an Anvil renderable - you can use `layoutTransition(new LayoutTransition())` property setter to enable layout transitions.

Another option would be to use fragments and use `setCustomAnimation` to the fragment transition. Not my favorite way, but some people really like Fragments.

Finally, you can write animations inside your Anvil renderables. There is `config(() -> { ... })` property setter allowing you to bind a lambda that will be executed only once (when the rendreable is added to the layout). In that lambda you may use `Anvil.currentView()` to get the instance of your current android view class. Then you may call `animate()` etc to animate the actual instance of the view whenever it's needed.

Summing up, with Anvil you can still use layout transitions, fragment transitions or view animations, as much as you would use it with pure android views inflated from XML.

I hope Anvil will work well for you, and feel free to ask if you have any questions.

**Comment by FunnyDevs** [comment]
Thanks a lot for the complete answer! I think best option, for my purpose, is the third. Now the moment has come to try Anvil.

**Comment by zserge** [comment]
Actually, sorry, I was wrong about `config()` function. It's should be `init(() -> { ... })` - the function that is executed once when the view is inflated.

`config` used to be an old name used in early versions of Anvil. `init` is supposed to be a more descriptive word because "initialization" happens only once, while "configuration" can theoretically happen multiple times. So `init` function is guaranteed to be run once for each view instance and it's a good place to store references to the real view objects, or call view methods to initialize it directly.
